[CI] StringsTests testDeleteAny failing

### CI Link

https://gradle-enterprise.elastic.co/s/323cgxy7itfky

### Repro line

REPRODUCE WITH: ./gradlew ':server:test' --tests "org.elasticsearch.common.StringsTests.testDeleteAny" -Dtests.seed=8A64C9DD086550D3 -Dtests.locale=it -Dtests.timezone=Asia/Baghdad -Druntime.java=17	

### Does it reproduce?

Yes

### Applicable branches

main

### Failure history

https://gradle-enterprise.elastic.co/scans/tests?search.timeZoneId=Europe/Rome&tests.container=org.elasticsearch.common.StringsTests&tests.test=testDeleteAny

### Failure excerpt

```
java.lang.AssertionError: |  
  | Expected: "짞ĐGⰣ*ﮢ" |  
  | but: was "짞ĐGⰣﮢ"
```

--- a/server/src/test/java/org/elasticsearch/common/StringsTests.java
+++ b/server/src/test/java/org/elasticsearch/common/StringsTests.java
@@ -180,7 +180,6 @@ public class StringsTests extends ESTestCase {
         );
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/91075")
     public void testDeleteAny() {
         assertNull(deleteAny((CharSequence) null, "abc"));
         assertNull(deleteAny((String) null, "abc"));

--- a/server/src/test/java/org/elasticsearch/common/StringsTests.java
+++ b/server/src/test/java/org/elasticsearch/common/StringsTests.java
@@ -191,10 +191,12 @@ public class StringsTests extends ESTestCase {
 
         String testStr = randomUnicodeOfLength(10);
         String delete = testStr.substring(testStr.length() - 1) + testStr.substring(0, 1);
-        assertThat(deleteAny(testStr, delete), equalTo(testStr.substring(1, testStr.length() - 1)));
-        assertThat(deleteAny(new StringBuilder(testStr), delete), hasToString(testStr.substring(1, testStr.length() - 1)));
-
-        // this method doesn't really work with surrogates
+        String expected = testStr.chars()
+            .mapToObj(Character::toString)
+            .filter(c -> delete.contains(c) == false)
+            .collect(Collectors.joining());
+        assertThat(deleteAny(testStr, delete), equalTo(expected));
+        assertThat(deleteAny(new StringBuilder(testStr), delete), hasToString(expected));
     }
 
     public void testSplitStringToSet() {

